src/assignment/backend/sql/role/create.rs

```diff
@@ -0,0 +1,145 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use sea_orm::DatabaseConnection;
+use sea_orm::entity::*;
+use serde_json::json;
+use uuid::Uuid;
+
+use crate::assignment::backend::error::{AssignmentDatabaseError, db_err};
+use crate::assignment::types::role::{Role, RoleCreate};
+use crate::db::entity::role as db_role;
+
+/// Create a new role
+pub async fn create(
+    db: &DatabaseConnection,
+    role: &RoleCreate, // ← Using RoleCreate instead of Role
```

Comment:
mostly the backends create methods take creation struct by value and not by reference